apps/cowswap-frontend/src/utils/orderUtils/getOrderPermitAmount.ts

```diff
@@ -0,0 +1,34 @@
+import { Erc20__factory } from '@cowprotocol/abis'
+import type { LatestAppDataDocVersion } from '@cowprotocol/app-data'
+import { COW_PROTOCOL_VAULT_RELAYER_ADDRESS, SupportedChainId } from '@cowprotocol/cow-sdk'
+import { BigNumber } from '@ethersproject/bignumber'
+
+import { ParsedOrder } from './parseOrder'
+
+const erc20Interface = Erc20__factory.createInterface()
+
+export function getOrderPermitAmount(chainId: SupportedChainId, order: ParsedOrder): BigNumber | null {
+  if (!order.fullAppData) return null
+
+  try {
+    const spenderAddress = COW_PROTOCOL_VAULT_RELAYER_ADDRESS[chainId].toLowerCase()
+    const appData: LatestAppDataDocVersion = JSON.parse(order.fullAppData)
+    const preHooks = appData.metadata.hooks?.pre
+
+    if (!preHooks) return null
+
+    const permitData = preHooks
+      .map((hook) => erc20Interface.decodeFunctionData('permit', hook.callData))
+      .find((decoded) => {
+        return (
+          decoded.spender?.toLowerCase() === spenderAddress &&
+          decoded.owner?.toLowerCase() === order.owner.toLowerCase() &&
+          (decoded.deadline as BigNumber)?.toNumber() > Date.now() / 1000
+        )
```

Comment:
I still don't get it. I think the try catch is outside of the map function. If you throw inside a map, it propagates the error up.

Did a small test, just in case:

<img width="573" alt="image" src="https://github.com/user-attachments/assets/d53b3325-d852-42c4-935d-b549bdc55794" />

Maybe I'm missing something, but shouldn't you have the try-catch inside the map function? Probably, if you want to do it like this, could be a reduce function which only adds it to the result if it succeeds with the decoding

Also, probably all these callsdata start with the same prefix, the name of the encoded name of the function permit. 
Wouldn't it be good to do some prefiltering so we don't try to decode other hooks?

Reply:
You are right, I'm sorry!
Fixed it